a10_nlbaas2oct/driver.py

```diff
@@ -434,22 +484,35 @@ def main():
                 if not device_info:
                     continue
                 device_name = device_info['name']
+                if CONF.migration.flavor_id_list and CONF.migration.lb_id_list:
+                    if fl_list_length < lb_list_length and not CONF.migration.default_flavor_id:
+                        for flavor_id in range(fl_list_length, lb_list_length):
+                            flavor = _migrate_flavor(LOG, a10_config, o_session, device_name, None)
+                            conf_flavor_id_list.insert(flavor_id, flavor)
+                            fl_list_length = fl_list_length + 1
                 if device_name != curr_device_name:
-                    fl_id = _migrate_flavor(LOG, a10_config, o_session, device_name)
+                    fl_id_list = _migrate_flavor(LOG, a10_config, o_session, device_name, conf_flavor_id_list)
                     curr_device_name = device_name
-                _migrate_slb(LOG, n_session, o_session, lb_id,
-                             fl_id, tenant_id, n_lb, CONF.migration.ignore_l7rule_status)
+                if fl_id_list:
+                    fl_id = fl_id_list[counter]
+                    counter = counter + 1
+                    _migrate_slb(LOG, n_session, o_session, lb_id,
+                                 fl_id, tenant_id, n_lb, CONF.migration.ignore_l7rule_status)
+
+                    # Rollback everything if we are in a trial run otherwise commit
+                    if CONF.migration.trial_run:
+                        o_session.rollback()
+                        n_session.rollback()
+                        LOG.info('Simulated ' + lb_success_msg + ' successful', lb_id)
+                    else:
+                        o_session.commit()
+                        n_session.commit()
+                        LOG.info('Successful ' + lb_success_msg, lb_id)
+                else:
+                    full_success_msg = "Skipping the migration as the specified flavor cannot be found"
```

Comment:
This is a failure and not a success message. If they didn't provide enough information for migration to occur then the migration failed. It should be logged as a warning.

Reply:
Also, if we skip the migration fully due to a failure like this then we shouldn't execute the clean_up slb flow. We also want to make sure the rollbacks occur on the objects that could have been affected by the migration.

Reply:
Added rollback for n_session and o_session and also added warning in this case.
As we shouldn't execute clean_up slb flow as well, returning in this case after giving the warning.